lib/services/services.go

```diff
@@ -52,9 +56,27 @@ func (s *ServiceRegistry) StartAll() {
 	s.logger.Debug("All services started.")
 }
 
+// PauseServices pauses key services before shutdown to allow a graceful shutdown
+func (s *ServiceRegistry) PauseServices() {
```

Comment:
Not sure why we need to introduce `Pause()` to the service interface if we just need to stop the sync and state services before we stop the others.  I would just modify `StopAll` to be less generic and stop the sync and state services first before stopping the rest.

Reply:
This is not possible as state service is a dependency of all the other services, so shutting it down anywhere but last causes a panic. This is what made this tricky in the first place. Introducing `Pause()` is a workaround to essentially stop the services without actually shutting it down

Reply:
Oh I see.  I would rather explicitly pause the sync and state services rather than introduce the `Pause` method for all services.  You can just access the `sync.Service` or `state.Service` via the map using `reflect.TypeOf` instead of ranging through each service.  You can then just cast specifically to `sync.Service` or `state.Service` which will have the `Pause` method.  That way you don't need to introduce it to every single `Service`.

Reply:
Oh nice, I like this idea. Will implement

Reply:
updated